### PR TITLE
Update CreateService

### PR DESCRIPTION
**Pull request by devgabrielcoman**
## JIRA
N/A

## Description
- Refactored the CreateServiceProvider into UsernameServiceProvider
- This now checks for username validity
- Removed the Protocol that was defined for CreateService in the Data SDK
- Added a VerifiedUsername model based on VerifiedUsernameModelProtocol
- Fixed tests
- Added tests for UsernameServiceProvider

### CODE REVIEW

File: Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+                done()
+                
+            }
+        }
+    }
+    
+    func testVerifyUsernameShortUsername() {
+        
+        let userName = "ib"
+        let request = VerifiyUsernameRequest.init(environment: self.environment,
```
Comment (me2k) [inline Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift]: VerifiyUsernameRequest typo here, Verify not Verifiy

File: Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        waitUntil { done in
+            
+            self.usernameService.verifiyUsername(username: userName) { (response, error) in
+                
+                expect(response).toNot(beNil())
+                expect(response?.available).to(beTrue())
+                expect(response?.allowed).to(beTrue())
+                
+                done()
+
```
Comment (Tbruni85) [inline Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift]: space

File: Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                expect(errorResponse).toNot(beNil())
+                
+                expect(errorResponse?.validationError).to(beTrue())
+                expect(errorResponse?.usernameValidationError).to(beTrue())
+                expect(errorResponse?.userBanned).to(beFalse())
+                expect(errorResponse?.usernameAlreadyTaken).to(beFalse())
+                expect(errorResponse?.usernameForbidden).to(beFalse())
+                
+                done()
+
```
Comment (Tbruni85) [inline Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift]: space

File: Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        waitUntil { done in
+            
+            self.usernameService.verifiyUsername(username: userName) { (response, error) in
+                
+                expect(response).toNot(beNil())
+                expect(response?.available).to(beTrue())
+                expect(response?.allowed).to(beFalse())
+                
+                done()
+
```
Comment (Tbruni85) [inline Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift]: space

File: Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        waitUntil { done in
+            
+            self.usernameService.verifiyUsername(username: userName) { (response, error) in
+                
+                expect(response).toNot(beNil())
+                expect(response?.available).to(beFalse())
+                expect(response?.allowed).to(beTrue())
+                
+                done()
+
```
Comment (Tbruni85) [inline Example/PopJamSDK_Tests/Username/Username+ObjectProviderTests.swift]: space

File: Example/PopJamSDK_Tests/Username/VerifiedUsername+MappingTests.swift [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    
+    func testVerifiedUsernameMappingFail() {
+        
+        let JSON = [:] as [String : Any]
+        
+        let verified = try? VerifiedUsername.decode(JSON)
+        
+        expect(verified).to(beNil())
+    }
+
```
Comment (Tbruni85) [inline Example/PopJamSDK_Tests/Username/VerifiedUsername+MappingTests.swift]: space

File: Example/PopJamSDK_Tests/Username/VerifiedUsernameTests.swift [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        expect(foo.hash).toNot(equal(bar.hash))
+    }
+    
+    func testObjcVerifiedUsernameNotEqualityWithDifferentClassAndSameIdentifier() {
+        let foo =  VerifiedUsername.init(available: true, allowed: true)
+        let bar = "12345"
+        
+        expect(foo.isEqual(bar)).to(beFalse())
+    }
+
```
Comment (Tbruni85) [inline Example/PopJamSDK_Tests/Username/VerifiedUsernameTests.swift]: space

**Pull request merged**